planet/clients/orders.py

```diff
@@ -271,18 +274,72 @@ async def download_asset(self,
                              progress_bar=progress_bar)
         return dl_path
 
+    @staticmethod  # should this be an async function?
+    def calculate_checksum(manifest_data: dict,
+                           filenames: list,
+                           checksum: str):
+        """Calculate checksum and validate that it passes.
+
+        Parameters:
+            manifest_data: The contents of the opened manifest.json file.
+            filenames: List of downloaded assets.
+            checksum: The type of checksum hash- 'MD5' or 'SHA256'.
+
+        Returns:
+            Message confirming checksums were succesful.
```

Comment:
@mkshah605 in my experience, having functions return message strings isn't very useful. And at line 380 we're not doing anything with the message. It would be fine for this method to `return None` (Python's default if there is no return statement) on success and raise an exception (as you do) on failure.

Also, I feel like the `calculate` in the name of the method implies that we are calculating checksums so that they would be returned and used by other code. That's not the case here, so how about `validate_checksums` instead?

Reply:
Agree! Will do.